test/studies/labelprop/labelprop-tweets.chpl

```diff
@@ -187,10 +186,34 @@ proc process_json(logfile:channel, fname:string, ref Pairs) {
 
   if progress then
     writeln(fname, " : processing");
- 
+
   while true {
-    got = logfile.readf("%~jt", tweet, error=err);
-    if got && !err {
+    try! {
+      got = logfile.readf("%~jt", tweet);
+    } catch e: SystemError {
+      if e.err == EEOF {
+        break;
+      } else if e.err == EFORMAT {
+        if verbose then
+            stdout.writeln("error reading tweets ", fname, " offset ",
+              logfile.offset(), " : ", errorToString(e.err));
+
+        // read over something else
+        try! {
+          got = logfile.readf("%~jt", empty);
```

Comment:
this one definately shouldn't be in a `try!` - instead, failure here should output the "severe error" message.
Maybe it should use a two-level try/catch, where the inner level handles EOF/read over something, and the outer level prints out the sever error message?